File: 1759-Count_Number_of_Homogenous_Substring/Count_Number_of_Homogenous_Substring.go
package countnumberofhomogenoussubstring

func countHomogenous(s string) int {
	mod := 1000000007
	var total int64 = 0
	var count int64 = 0
	curr := s[0]

	for i, _ := range s {
		if s[i] == curr {
			count++
		} else {
			count = 1
			curr = s[i]
		}
		total += count
	}

	return int(total % int64(mod))
}

File: 1759-Count_Number_of_Homogenous_Substring/Count_Number_of_Homogenous_Substring.java
class Solution {
    public int countHomogenous(String s) {
        char c = s.charAt(0);
        int count = 0;
        long totalCount = 0l;
        
        
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == c) {
                count++;
            }
            else {
                count = 1;
                c = s.charAt(i);
            }
            totalCount += (long)count;
        }
        
        return (int)(totalCount % 1000000007);
    }
}